### PR TITLE
fix(prover): fix degrees logic in prover new

### PR DESCRIPTION
**Pull request by amoylan2**
### Purpose or design rationale of this PR

This is a complement for https://github.com/scroll-tech/scroll/pull/1509

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [ ] feat: A new feature
- [x] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [ ] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

**Review by lispc — DISMISSED** [review]

File: prover/src/zk_circuits_handler/darwin_v2.rs [hunk 1]
```diff
@@ -99,6 +88,21 @@ impl DarwinV2Handler {
         Ok(handler)
     }
 
+    fn get_degrees(prover_types: &std::collections::HashSet<ProverType>) -> Vec<u32> {
+        fn get_degrees_inner(prover_type: &ProverType) -> Vec<u32> {
+            match prover_type {
+                ProverType::Chunk => prover_darwin::config::ZKEVM_DEGREES.clone(),
```
Comment (roynalnaruto) [inline prover/src/zk_circuits_handler/darwin_v2.rs]: `prover_darwin_v2` here

**Review by roynalnaruto — CHANGES_REQUESTED** [review]
We should:
- Move `get_degrees` to the `common` module
- Define `get_degrees_inner` in each of the `darwin` and `darwin_v2` modules
- Fix `darwin_v2`'s impl where you have still used `prover_darwin::config` (should be `prover_darwin_v2::config`

File: prover/src/zk_circuits_handler/darwin_v2.rs (hunk 1 above)
Comment (roynalnaruto) [reply prover/src/zk_circuits_handler/darwin_v2.rs]: In the `common::get_degrees` you can receive the `get_degrees_inner` method as an argument and supply it from each zk circuits handler.

Comment (amoylan2) [reply prover/src/zk_circuits_handler/darwin_v2.rs]: thanks, it makes sense.

Comment (amoylan2) [reply prover/src/zk_circuits_handler/darwin_v2.rs]: Changes made, pls have a look again

**Review by roynalnaruto — APPROVED** [review]
Looks good!

**Review by lispc — APPROVED** [review]

**Pull request merged**